Core/Graphics/Include/Shared/Babylon/Graphics.h

```diff
@@ -1,18 +1,17 @@
 #pragma once
 
 #include <Babylon/JsRuntime.h>
+#include <Babylon/GraphicsPlatform.h>
```

Comment:
Now everything that consumes Graphics will consume platform headers, is this your intention?

Reply:
That's the idea, though perhaps I've misunderstood something. Graphics.h exports the public contract of the Graphics component, which by definition (now that we've introduced `GraphicsConfiguration` contains platform-specific information. Anyone who wants consume the public Graphics contract should have to include Graphics.h, and only Graphics.h. But this public contract has nothing to do with the internal contract, which has many different pieces, only some of which are platform specific. Internal contract consumers, generally speaking, should not include Graphics.h at all. (They can -- it's possible -- but I don't believe anyone has reason to in the current repo, and I suspect I would regard any component that wanted to consume both the public and internal contracts of Graphics simultaneously as a "code smell" indicative of an architecture problem, most likely that the consumer was really two concepts being mistakenly implemented as one.) Instead, consumers of the internal contract should include only the parts of the internal API that they need, typically this will be just GraphicsImpl.h; and those who also need GraphicsPlatform.h are, by virtue of the fact that they need that, certainly doing something platform-specific already.

That's the idea, anyway; let's definitely follow up if I've missed something, though.

Reply:
I see, I definitely needed to see this to fully contextualize what you were going for, thanks for the explanation.

Reply:
This is kind of a side note, but we shouldn't depend on the include path to include dependent headers. This requires that our consumers use include directories that might not be what they want to do. Anyways, something to keep in mind. If we ever create a packaged solution for Babylon Native, this will become an issue.